packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -203,20 +139,132 @@ class DirectStripeRoutes {
   }
 
   /**
-   * Verify that the invoice was paid successfully.
+   * Create Subscription for New Customer
+   *
+   * @param {string} uid
+   * @param {string} email
+   * @param {string} displayName
+   * @param {string} paymentToken
+   * @param {AbbrevPlan} selectedPlan
+   *
+   * @returns {Promise<Subscription>}
+   */
+  async createSubscriptionNewCustomer(
+    uid,
+    email,
+    displayName,
+    paymentToken,
+    selectedPlan
+  ) {
+    let customer;
+
+    try {
+      customer = await this.stripeHelper.createCustomer(
+        uid,
+        email,
+        displayName,
+        paymentToken
+      );
+    } catch (err) {
+      if (err.type === 'StripeCardError') {
+        throw error.rejectedSubscriptionPaymentToken(err.message, err);
+      }
+      throw err;
+    }
+
+    const subscription = await this.stripeHelper.createSubscription(
```

Comment:
Might be good to move the `createSubscription` call into the try/catch along with `createCustomer`? I still have a strong suspicion that this `StripeCardError` will come up from either of these spots

Reply:
I was in debate over this very thing: to put it all in the `try/catch` or not